Fixes #22373 - Audit manifest import/refresh using Organization

kgaikwad opened the pull request.

kgaikwad commented:
@ares,
This PR includes audit changes for import/refresh manifest using Organization.

kgaikwad commented on test/actions/katello/organization_test.rb:
```diff
@@ -130,6 +130,23 @@ class ManifestRefreshTest < TestBase
+      action.stubs(:exclusive_lock!)
+
+      plan_action(action, acme_org)
+      assert_difference 'Audit.count', 1 do
```
Note - when I have added `acme_org.audits`, it was giving  `<ActiveRecord::Associations::CollectionProxy []>` whereas in `Audit`  getting a new record for Organization.

ares commented on app/lib/actions/katello/organization/manifest_import.rb:
```diff
@@ -26,6 +26,13 @@ def plan(organization, path, force)
           _("Import Manifest")
         end
+
+        def finalize
```
this task does not call `plan_self` in its `plan` method, therefore finalize is not triggered, which explains why I don't see the audit being added or refreshed_at being updated. simply adding plan_self to the end of sequence block should fix it

kgaikwad replied:
Yes, right it doesn't trigger finalize. My mistake that I have tested test cases and I wasn't able to test full flow through UI as on dev setup faced some different issue.
I will fix it asap.

ares commented on app/lib/actions/katello/organization/manifest_refresh.rb:
```diff
@@ -42,6 +42,13 @@ def rescue_strategy
           _("Refresh Manifest")
         end
+
+        def finalize
```
this also does not call plan_self in `plan` method, so finalize is never triggered

kgaikwad commented on db/migrate/20180218191449_add_refreshed_at_to_organization.rb:
```diff
@@ -0,0 +1,9 @@
+class AddRefreshedAtToOrganization < ActiveRecord::Migration[5.1]
+  def up
+    add_column(:taxonomies, :refreshed_at, :datetime)
```
Yes, I agree.

kgaikwad commented:
@ares,
Thank you for reviewing a pull-request.
I have corrected code as per review comments.
I was able to test manifest import & refresh actions now I can see audit records.

Added audit changes to `Actions::Katello::Organization::ManifestDelete`. Test case is running fine but when I tried to delete manifest from web-ui, created task for it. That task was running for 3-4 hours and on console, it was continuously showing below request logs. Afterwards, I forcefully stopped my server.

![screenshot from 2018-02-24 18-57-35](https://user-images.githubusercontent.com/6470528/36631214-224116e8-199a-11e8-9fda-0d8425ebf659.png)

jlsherrill commented:
This all works as expected, i did notice its hard to tell the difference between upload/refresh/delete:

![audit1](https://user-images.githubusercontent.com/395077/36923322-eb336d38-1e38-11e8-84ac-54f5188cc3a4.png)

If you click on it you see more info:

![audit2](https://user-images.githubusercontent.com/395077/36923332-f2844c9c-1e38-11e8-9114-7982daa7a092.png)

@ares i guess this is the best we can do with the current auditing system? or do you have some other suggestion?

ares commented:
Sadly that's all we can do for now. It would be great to have ability to set custom label but so far we can only set comments which are displayed on audit detail page.
